### PR TITLE
Updating README.md and adding a d20 feature list

### PR DESCRIPTION
**Pull request by SebaLukas**
## Describe your changes
Updating README.md.
Adding a d20 feature list to show the current d20 feature support.

## Issue ticket number and link
Without a feature list, it is quite difficult to get an overview of which D20 feature this library already contains.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,45 @@
 ISO 15118 library suite
 =======================
 
-Currently, this is a proof of concept library.
+This is a C++ library implementation of ISO 15118 for part 20.
 
-Integration of ISO 15118 development for EVerest is under heavy development
-currently. Many features are already implemented, many are still to follow.
+Integration of ISO15118-20 development for EVerest is under heavy development
+currently.
```
Comment (Pietfried) [inline README.md:7]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1,23 +1,45 @@
...
+This is a C++ library implementation of ISO 15118 for part 20.
 
-Integration of ISO 15118 development for EVerest is under heavy development
-currently. Many features are already implemented, many are still to follow.
+Integration of ISO15118-20 development for EVerest is under heavy development
+currently. 
+
+## ISO 15118-20 Support
+
+The following tables show the current support for the listed EVSE ISO15118-20 features.
```
Comment (Pietfried) [inline README.md:11]:
```suggestion
The following table shows the current support for the listed EVSE ISO15118-20 features.
```

File: README.md [hunk 3]
```diff
@@ -1,23 +1,45 @@
...
+| Dynamic Mode (+ MobilityNeedsMode) | :heavy_check_mark: |
+| Private Env                        |                    |
+| Pause/Resume & Standby             | WIP                |
+| Schedule renegotation              |                    |
+| Smart Charging                     |                    |
+| Multiplex messages                 |                    |
+| Internet service                   |                    |
+| Parking status service             |                    |
 
-But: This repository here is not the place where implementation happens.
```
Comment (Pietfried) [inline README.md:9]: I would use something like

`ISO15118 support is distributed accross multiple repositories and modules in EVerest. Please see the following references of other ISO15118 related development:`

File: README.md [hunk 4]
```diff
@@ -1,23 +1,45 @@
 ISO 15118 library suite
 =======================
 
-Currently, this is a proof of concept library.
+This is a C++ library implementation of ISO 15118 for part 20.
```
Comment (Pietfried) [inline README.md:4]:
```suggestion
This is a C++ library implementation of ISO 15118-20 and ISO15118-2. The implementation of ISO15118-20 is currently under heavy development. ISO15118-2 will follow and is currently covered by the [EvseV2G module](https://github.com/EVerest/everest-core/tree/main/modules/EvseV2G) of everest-core.
```

File: README.md [hunk 5]
```diff
@@ -1,23 +1,45 @@
...
+
+The following tables show the current support for the listed EVSE ISO15118-20 features.
+
+| Feature                            | Supported          |
+|------------------------------------|--------------------|
+| TCP, TLS 1.2 & 1.3                 | :heavy_check_mark: |
+| DC, DC_BPT                         | :heavy_check_mark: |
+| AC, AC_BPT                         | WIP                |
+| WPT                                |                    |
+| ACDP                               |                    |
```
Comment (Pietfried) [inline README.md:19]: Maybe we can provide the full name of some abbreviations as well. What do you think?

File: README.md (hunk 3 above)
Comment (Pietfried) [reply README.md:9]: You can then also drop this last sentence:

`This is just a short overview to show...`

Comment (SebaLukas) [reply README.md:9]: :+1:

File: README.md (hunk 5 above)
Comment (SebaLukas) [reply README.md:19]: :+1: I will add a list of abbreviations under this table

File: README.md [hunk 6]
```diff
@@ -1,35 +1,60 @@
...
+| Internet Service                   |                    |
+| Parking Status Service             |                    |
+
+### List of Abbreviations
+
+| Abbreviation | Description                            |
+|--------------|----------------------------------------|
+| ACDP         | automatic connection device pantograph |
+| BPT          | bidirectional power transfer           |
+| WPT          | wireless power transfer                |
```
[the remark this answers is not in the document]
Comment (Pietfried) [reply README.md:37]: That was added based on [my comment](https://github.com/EVerest/libiso15118/pull/48#discussion_r1842627546). Maybe people who know ISO15118 better dont need this indeed, so its also fine for me to drop it

[the remark this answers is not in the document]
Comment (SebaLukas) [reply README.md]: Removed it

**Review by a-w50 — APPROVED** [review]
:100:

**Pull request merged**